biohansel/create/schema_generator.py

```diff
@@ -0,0 +1,137 @@
+        record_dict: dictionary of records obtained from the reference sequence file
+
+    Returns:
+        df_list: list of DataFrames that contain snv sequences related to each chromosome for that particular cluster group
+    """
+
+    df_list = []
+    curr_df.CHROM = curr_df.CHROM.str.strip()
+    unique_chromosomes = curr_df.CHROM.unique()
+    for chromosome in unique_chromosomes:
+        record_seq = record_dict[chromosome]
+        max_sequence_value = len(record_seq)
+        sequences = str(record_seq)
+        curr_chrom_df = curr_df[curr_df['CHROM'].str.match(
+            chromosome.strip())]
+
+        positions = pd.Series(curr_chrom_df.index, index=curr_chrom_df.index)
+        ref_seqs = positions.apply(
+            get_subsequences,
+            args=(sequences, sequence_length, max_sequence_value))
+        alt_seqs = ref_seqs.str.slice(
+            0, sequence_length) + curr_chrom_df.ALT + ref_seqs.str.slice(
+            sequence_length + 1, sequence_length + sequence_length + 1)
+
+        curr_chrom_df['ref_sequences'] = ref_seqs
+        curr_chrom_df['alt_sequences'] = alt_seqs
+        df_list.append(curr_chrom_df)
+
+    return df_list
+
+def get_subsequences(position: int, seq: str, sequence_length: int,
+                     max_sequence_value: int) -> str:
+    """Get the sequences that are before are after the specified SNV
+    Args:
+        position: the position of the SNV based off of the reference genome
+        seq: the set of sequences from the reference genome
+        sequence_length: the amount of upstream and downstream sequences added to the SNV and included in the schema
+                        output
+        max_sequence_value: the length of the reference genome
+
+    Returns:
+        sequence_sequence: the sequence that is found at the specified genome positions from the reference genome
+
+    """
+
+    specific_sequence = seq[max(0, position - (sequence_length + 1)):min(
+        max_sequence_value, position + sequence_length)]
+
+    return specific_sequence
+
+
+def get_sequence_string(ratio_value: int, chromosome: str, position: int, group: str, reference_snv: str,
+                        alternate_snv: str) -> str:
+    """Creates the string output for the schema file
+    Args:
+        ratio_value: the ratio_value for that particular snv, with 1 being that all genomes contain the snv and 0 being
+                    none of the genomes contain the sequence
+        chromosome: the current chromosome of the reference genome sequence
+        position: the position of the SNV
+        group: the group membership of the SNV
+        reference_snv: the genome sequence of the reference genome
+        alternate_snv: the genome sequence of the alternate form of the SNV
+
+    Returns:
+        sequence_string: the sequence string to be outputted into the schema file
+
+    """
+
+    return (textwrap.dedent(f"""\
```

Comment:
Please remove the `textwrap.dedent` call and either convert this to a multiline string that doesn't have leading whitespace or into a multiple single line f-strings e.g.

```python
    return f""">({chromosome}){position}-{group}
{alternate_snv if ratio_value > 0 else reference_snv}
>negative({chromosome}){position}-{group}
{reference_snv if ratio_value > 0 else alternate_snv}
"""

# OR

    return (f">({chromosome}){position}-{group}"
            f"{alternate_snv if ratio_value > 0 else reference_snv}"
            f">negative({chromosome}){position}-{group}"
            f"{reference_snv if ratio_value > 0 else alternate_snv}")
```

Reply:
yes this has been addressed in 46cbc1f527bbf6d0225c3fdb51585fbe19ffc4f7